### ISSUE TITLE
Agent unit tests fail on go 1.5.1

### ISSUE THREAD

**Issue by yatzek** [cloudfoundry/bosh#967]
```
[1443617288] Agent Suite - 73/73 specs ••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••• SUCCESS! 37.690825ms PASS
[1443617288] Action Suite - 207/207 specs •••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••[1443617288] Alert Suite - 22/22 specs •••••••••••••••••••••• SUCCESS! 1.653709ms PASS
[1443617288] Applier Suite - 20/20 specs •••••••••••••••••••• SUCCESS! 1.433155ms PASS
[1443617288] Applyspec Suite - 18/18 specs •••••••••••••••••• SUCCESS! 4.722352ms PASS
[1443617288] Bundlecollection Suite - 34/34 specs •••••••••••••••••••••••••••••••••• SUCCESS! 2.987718ms PASS
[1443617288] Jobs Suite - 49/49 specs ••••••••••••••••••••••••••••••••••••••••••••••••• SUCCESS! 5.290674ms PASS
[1443617288] Models Suite - 4/4 specs •••• SUCCESS! 437.425µs PASS
[1443617288] Packages Suite - 37/37 specs ••••••••••••••••••••••••••••••••••••• SUCCESS! 3.76521ms PASS
[1443617288] Cmdrunner Suite - 18/18 specs •••••••••••••••••• SUCCESS! 2.428006ms PASS
[1443617288] Compiler Suite - 19/20 specs ••••P••••••••••••••• SUCCESS! 2.467223ms PASS
[1443617288] Drain Suite - 22/22 specs •••••••••••••••••••••• SUCCESS! 1.651573ms PASS
[1443617288] Script Runner Suite - 6/6 specs •••••• SUCCESS! 3.017743ms PASS
[1443617288] Task Suite - 22/22 specs •••••••••••••••••••••• SUCCESS! 2.055374218s PASS
[1443617288] AgentClient Suite - 5/5 specs ••••• SUCCESS! 449.231µs PASS
[1443617288] Applyspec Suite - 1/1 specs • SUCCESS! 839.54µs PASS
[1443617288] HTTP AgentClient Suite - 41/41 specs ••••••••••••••••••••••••••••••••••••••••• SUCCESS! 6.986999ms PASS
[1443617288] App Suite - 23/23 specs ••••••••••••••••••••••• SUCCESS! 11.602666ms PASS
[1443617288] Handler Suite - 8/8 specs •••••••• SUCCESS! 709.488µs PASS
[1443617288] Httpsdispatcher Suite - 8/9 specs ••P•••2015/09/30 13:49:12 http: TLS handshake error from 127.0.0.1:43539: tls: no cipher suite supported by both client and server
•2015/09/30 13:49:14 http: TLS handshake error from 127.0.0.1:43569: tls: no cipher suite supported by both client and server
•• SUCCESS! 16.248254309s PASS
[1443617288] Infrastructure Suite - 117/117 specs ••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••••• SUCCESS! 28.200911ms PASS
testing: warning: no tests to run
PASS
[1443617288] Devicepathresolver Suite - 25/25 specs ••••••••••••••••••••••••• SUCCESS! 4.751753138s PASS
[1443617288] Jobsupervisor Suite - 33/33 specs ••••••••••••2015/09/30 13:49:27 mail from: "sender@example.org"
•2015/09/30 13:49:27 mail from: "sender@example.org"
•••••••••••••••••••• SUCCESS! 8.714792ms PASS
[1443617288] Monit Suite - 21/21 specs •
------------------------------
• Failure [1.109 seconds]
MonitRetryStrategy Try when all responses are only 503s [It] retries until maxUnavailableAttempts + maxOtherAttempts are exhausted 
/home/jacek/workspace/cloudfoundry/bosh/go/src/github.com/cloudfoundry/bosh-agent/jobsupervisor/monit/monit_retry_strategy_test.go:76

  Timed out after 1.000s.
  Expected
      <int>: 1
  to equal
      <int>: 0

  /home/jacek/workspace/cloudfoundry/bosh/go/src/github.com/cloudfoundry/bosh-agent/jobsupervisor/monit/monit_retry_strategy_test.go:238
------------------------------
••
------------------------------
• Failure [1.027 seconds]
MonitRetryStrategy Try when the initial attempt is a non-503 error [It] retries for maxOtherAttempts 
/home/jacek/workspace/cloudfoundry/bosh/go/src/github.com/cloudfoundry/bosh-agent/jobsupervisor/monit/monit_retry_strategy_test.go:140

  Timed out after 1.000s.
  Expected
      <int>: 1
  to equal
      <int>: 0

  /home/jacek/workspace/cloudfoundry/bosh/go/src/github.com/cloudfoundry/bosh-agent/jobsupervisor/monit/monit_retry_strategy_test.go:238
------------------------------
••••••••••••••••

Summarizing 2 Failures:

[Fail] MonitRetryStrategy Try when all responses are only 503s [It] retries until maxUnavailableAttempts + maxOtherAttempts are exhausted 
/home/jacek/workspace/cloudfoundry/bosh/go/src/github.com/cloudfoundry/bosh-agent/jobsupervisor/monit/monit_retry_strategy_test.go:238

[Fail] MonitRetryStrategy Try when the initial attempt is a non-503 error [It] retries for maxOtherAttempts 
/home/jacek/workspace/cloudfoundry/bosh/go/src/github.com/cloudfoundry/bosh-agent/jobsupervisor/monit/monit_retry_strategy_test.go:238

Ran 21 of 21 Specs in 2.710 seconds
FAIL! -- 19 Passed | 2 Failed | 0 Pending | 0 Skipped --- FAIL: TestMonit (2.71s)
FAIL

Ginkgo ran 25 suites in 1m23.28768578s
Test Suite Failed

 Running build script to confirm everything compiles...

SUITE FAILURE (exit code: 1)

 Checking with golint...
find: warning: you have specified the -maxdepth option after a non-option argument -type, but options are not positional (-maxdepth affects tests specified before it as well as those specified after it). Please specify options before other arguments.


 Govetting

 All good
rake aborted!
Command failed with status (1): [cd go/src/github.com/cloudfoundry/bosh-age...]
/home/jacek/workspace/cloudfoundry/bosh/bosh-dev/lib/bosh/dev/tasks/spec.rake:113:in `block (3 levels) in <top (required)>'
Tasks: TOP => spec:unit => spec:unit:agent
(See full trace by running task with --trace)

```

❯ go version                                                                                                   ⏎
go version go1.5.1 linux/amd64

**Comment by cppforlife** [comment]
I can confirm I also saw this problem.

**Comment by cppforlife** [comment]
i believe this issue was fixed some time ago once we switched to go 1.5.
